src/js/background.js

```diff
@@ -0,0 +1,214 @@
+const setLoadingState = async (active, tabId) => {
+  const [calExist] = await browser.tabs.executeScript(tabId, {
+    code: `document.querySelector(".cal-loading") !== null;`,
+  });
+
+  if (!calExist) {
+    await browser.tabs.executeScript(tabId, {
+      file: "/src/js/createLoading.js",
+    });
+    await browser.tabs.insertCSS(tabId, { file: "/src/css/createLoading.css" });
+  }
+
+  await browser.tabs.executeScript(tabId, {
+    code: `document.querySelector(".cal-loading").style.display = ${active} ? "flex" : "none";`,
+  });
+};
+
+const sendLogMessage = async (message, tabId) => {
+  await browser.tabs.executeScript(tabId, {
+    code: `console.log("[Chess.com analyse at lichess]: ${message}");`,
+  });
+};
+
+const fetchJSON = async (url, tabId) => {
+  try {
+    const response = await fetch(url);
+    if (!response.ok)
+      throw new Error("Error when attempting to fetch resource.");
+    return response.json();
+  } catch (error) {
+    sendLogMessage(error, tabId);
+    return null;
+  }
+};
+
+/*
+ * Portable Game Notation (PGN) is a standard plain text format for
+ * recording chess games, which can be read by humans and is also
+ * supported by most chess software
+ */
+const getPGN = async (playerName, gameUrl, tabId) => {
+  try {
+    const archives = (
+      await fetchJSON(
+        `https://api.chess.com/pub/player/${playerName}/games/archives`,
+        tabId
+      )
+    ).archives;
+    /*
+     * Archives are given from the oldest to the most recet games.
+     * Thus iterating from the last one (the most recent) is usually
+     * better, because players usually want to analyse their most recent
+     * games. You will probably make less fetches
+     */
+    for (let i = archives.length - 1; i >= 0; i--) {
+      const games = (await fetchJSON(archives[i], tabId)).games;
+      for (let j = games.length - 1; j >= 0; j--) {
+        if (games[j].url === gameUrl) {
+          return games[j].pgn;
+        }
+      }
+    }
+  } catch (error) {
+    sendLogMessage(error, tabId);
+    return null;
+  }
+
+  return null;
+};
+
+const getPGNManual = async (tabId) => {
+  await waitAndClick(".share");
+  await waitAndClick(".board-tab-item-underlined-component");
+  await waitAndClick(".share-menu-tab-pgn-toggle");
+
+  if (await waitForElement("[name='pgn']")) {
+    const [pgn] = await browser.tabs.executeScript(tabId, {
+      code: `document.querySelector("[name='pgn']").value;`,
+    });
+    await waitAndClick("[data-cy='share-menu-close']");
```

Comment:
Shouldn't these `waitFor...` get `tabId` as an argument?

You could add a `// @ts-check` directive at the beginning of the file to see TypeScript errors for this file. Perhaps that will be useful to get validation messages like omitting an argument from a function. Even though this is a JS file, you can still enhance TypeScript server's understanding of it by adding JSDoc comments, like [`@param`](https://jsdoc.app/tags-param.html) to describe the type of function parameters. I recommend checking it out